src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -529,85 +527,52 @@ public virtual void WriteLiteralTo([NotNull] TextWriter writer, string value)
             }
             else
             {
-                for (var i = 0; i < values.Length; i++)
+                if (values.Length == 1 && values[0].Prefix == "" &&
+                    (values[0].Value.Value is bool || values[0].Value.Value == null))
                 {
-                    var attrVal = values[i];
-                    var val = attrVal.Value;
-                    var next = i == values.Length - 1 ?
-                        suffix : // End of the list, grab the suffix
-                        values[i + 1].Prefix; // Still in the list, grab the next prefix
+                    var attributeValue = values[0];
+                    var positionTaggedAttributeValue = values[0].Value;
 
-                    if (val.Value == null)
+                    if (positionTaggedAttributeValue.Value == null || !(bool)positionTaggedAttributeValue.Value)
                     {
-                        // Nothing to write
-                        continue;
+                        // The value is null or just the bool 'false', don't write anything.
+                        return;
                     }
 
-                    // The special cases here are that the value we're writing might already be a string, or that the
-                    // value might be a bool. If the value is the bool 'true' we want to write the attribute name
-                    // instead of the string 'true'. If the value is the bool 'false' we don't want to write anything.
-                    // Otherwise the value is another object (perhaps an HtmlString) and we'll ask it to format itself.
-                    string stringValue;
+                    WritePositionTaggedLiteral(writer, prefix);
 
-                    // Intentionally using is+cast here for performance reasons. This is more performant than as+bool?
-                    // because of boxing.
-                    if (val.Value is bool)
+                    // The value is just the bool 'true', write the attribute name instead of the string 'true'.
+                    WriteAttributeValue(writer, attributeValue, name, suffix.Position);
+                }
+                else
+                {
+                    WritePositionTaggedLiteral(writer, prefix);
```

Comment:
The decision was to not write prefix only if the value contains `@false` or `@null` with no surrounding whitespace.